# Introduction

Bulk RNA-sequencing (RNA-seq) is a widely deployed assay that captures the gene expression of a group of cells.
Its ability to capture gene expression directly from cells has made it a flexible technology that has been adopted across diverse scientific fields, from XX to XX.
Due to the open science movement and funding mandates, the majority of the raw sequencing data produced from these experiments is deposited in public databases like the Sequence Read Archive, making these data sets available to be re-analyzed in new studies.
To date, XXX Tb/Pb of RNA-seq data is available in the SRA (CITE).
However, while these data sets are publicly available, they data exist in raw formats that must be re-processed and are difficult to access in aggregate. 
Further, the metadata accompanying each data set is often incomplete or inaccurate, making data sets of interest difficult to find (CITE: SRA metadata).

+ Human compendia and key insights
    + Recount/2/3
        + methods development (transfer learning model, sophie, )
        + gene network inference
    + GTEX [@doi:10.1038/ng.2653] (not the best consortium bc has genomic variants)
    + TCGA [@doi:10.1038/ng.2764]]
        + improved taxonomy of cancer types and subtypes [@doi:10.1016/j.cell.2018.03.042]
        + catalog of genomic and molecular changes associated with specific types of cancer [@doi:10.1016/j.cell.2018.03.042]
        + conserved and divergent oncogenic processes and pathways involved in the development of tumors [@doi:10.1016/j.cell.2018.03.042; @doi:10.1038/ng.2764]
    
A recurrent theme across previously generated compendia is the idea that a compendium and any published analyses applied to it should act as "launching pads" [quoted from TCGA value] for continued analysis of those data products, or that new analyses could act as rapid and resource-conserving hypothesis-generating steps for new research.
The compendia themselves become annotated resources that help define current blind spots in biology, helping to better distribute research resources to tackle those blind spots [@doi:10.1016/j.cell.2018.03.042]

+ Bacterial papers that need to be cited in the intro:
    + LS Trap
    + Colombos
    + refine.bio
    + TCOL