# Methods

## Selection and acquisition of RNA-seq samples for compendium

We identified isolate bacterial and archaeal Illumina RNA-seq samples on the Sequence Read Archive (SRA) using the search criteria: ("Bacteria"[Organism] OR "Bacteria Latreille et al. 1825"[Organism] OR bacteria[All Fields]) AND "bacteria"[orgn] AND ("biomol rna"[Properties] AND "platform illumina"[Properties] AND "filetype fastq"[Properties]) (query on April 7, 2022).
We filtered the set of experiments returned to Library Source = TRANSCRIPTOMIC, Library Strategy = RNA-Seq or OTHER, and the library selection to cDNA, other, RANDOM, unspecified, RT-PCR, RANDOM PCR, or PCR.  
We removed experiment accessions that had no public data (SRP345944, SRP305697, SRP235317, SRP234344, SRP234343, SRP234342, SRP234341, SRP234340, SRP234339, SRP234338, SRP234337, SRP234336, SRP234335, SRP234334, SRP234333, SRP234332, SRP234331, SRP234330, SRP234329, SRP234328, SRP234327, SRP234326, SRP234325, SRP234324, SRP234323, SRP234322, SRP234321, SRP234320, SRP234319, SRP234318, SRP231538, SRP231537, SRP231536, SRP231535, SRP231534, SRP231533, SRP231129, SRP231128, SRP231127, SRP231126, SRP231120, SRP231119, SRP231118, SRP231117, SRP231035, SRP231034, SRP231033, SRP231032, SRP231031, SRP231030, SRP231029, SRP231028, SRP231027, SRP231026, SRP231025, SRP231024, SRP231023, SRP231022, SRP231021, SRP231020, SRP231019, SRP231018, SRP231017', SRP231016, SRP231015, SRP231014, SRP231013, SRP231012, SRP231011, SRP231010, SRP231009, SRP230142, SRP230135, SRP230133).
We then used the `sra_ids_to_runinfo.py` script available at github.com/nf-core/fetchngs to retrieve metadata for each remaining experiment from the European Nucleotide Archive (ENA) [@doi:10.5281/zenodo.5070524; @doi:10.1093/nar/gkaa1028].
We retained Run Accessions with greater than 1 million reads, and an average read length longer than 31 base pairs. 

We retrieved single- or paired-end FASTQ files using the ENA FTP download link.
We used fastp with default settings to quality control the reads [@doi:10.1093/bioinformatics/bty560]. 
To combine paired-end reads into a single file, reads were interleaved after quality control.

## Determination of taxonomic composition of RNA-seq samples

We used sourmash gather to determine the taxonomic composition of each RNA-seq sample [@doi:10.1101/2022.01.11.475838].
We first generate FracMinHash sketches of each RNA-seq sample using `sourmash sketch` (`-p k=21,k=31,k=51,scaled=1000,abund`).
We then compared each RNA-seq sketch against sourmash databases for the Genome Taxonomy Database (GTDB) rs207 (URL), human transcriptome (https://osf.io/anj6b/), and Eukaryotic transcriptomes (https://osf.io/qk5th/) using `sourmash gather` (`--scaled 2000 -k 31`).
This produced a CSV with genomes from these databases that were detected in the RNA-seq sample.
We used this information to select the best reference species and to estimate contamination in the RNA-seq sample.

To select the best reference species, we first joined the `sourmash gather` results to a lineage table that detailed the taxonomic lineage of each genome in GTDB rs207.
For each RNA-seq sample, we selected the genome with the highest fraction of abundance-weighted k-mer overlap with query (`f_unique_weighted`).
We determined the species of this genome in GTDB, and assigned the RNA-seq sample this species reference.
We excluded any Eukaryotic species from this step, as we reasoned that RNA-seq samples on the SRA that were labelled as bacterial or archaeal likely targeted species from these kingdoms, and any Eukaryotic sequence that was captured likely represented contamination.

## Construction of species-level reference pangenomes

+ note used charcoal-cleaned genomes

## RNA-seq quantification

## Interconversion between annotation formats

This pipeline supports interconversion between gene- and ortholog-level annotations.
During the construction of the species-level pangenome, each genome is first annotated with Bakta (CITE).
Bakta produces an GFF annotation file that records the name of the contiguous sequence in a genome (unchanged from the original genome downloaded from GenBank), an assigned identifier for the predicted feature, the identity of the predicted feature (e.g. CDS, gene), and the start and end base pair coordinate of the predicted feature.
We use the assigned identifier to unite genome-level annotations assigned by Bakta first to the representative sequence in the 95% identity clusters, and then to ortholog annotations produced by eggnog (CITE). 
This provides interconversion between genome-level annotations and ortholog-level annotations.
To support interconversion with specific reference genome annotations (e.g. *Pseudomonas aeruginosa* strain PAO1 [GCF_000006765.1] annotations maintained by pseudomonas.com (CITE)), we use exact matching of contiguous sequence names and start and end base-pair coordinates between GFF annotation files predicted by Bakta and canonical reference genome annotations.
This means that annotations can only be transferred from the Bakta annotation to the canonical reference genome annotation if the reference genome is contained in the GTDB rs207 database used to build the pangenomes.
Through the pangenome gene cluster membership, specific reference genome annotations could be back-propagated to other genomes of the same species.
